tests/test_source.py

```diff
@@ -0,0 +1,101 @@
+from sys import version_info
+from typing import Type
+
+if version_info.major == 3 and version_info.minor < 8:
```

Comment:
There's this other PR, which raised the python version to 3.8.

Reply:
I'll rebase and drop the compat-nonsense :)